include/inexor/vulkan-renderer/render_graph.hpp

```diff
@@ -185,23 +177,29 @@ class RenderStage : public RenderGraphObject {
     RenderStage &operator=(const RenderStage &) = delete;
     RenderStage &operator=(RenderStage &&) = delete;
 
-    /// @brief Specifies that this stage writes to `resource`
-    void writes_to(const RenderResource &resource);
+    /// @brief Specifies that this stage writes to `resource`.
+    void writes_to(const RenderResource *resource);
 
-    /// @brief Specifies that this stage reads from `resource`
-    void reads_from(const RenderResource &resource);
+    /// @brief Specifies that this stage reads from `resource`.
+    void reads_from(const RenderResource *resource);
 
-    /// @brief Binds a descriptor set layout to this render stage
-    /// @note This function will soon be removed
+    /// @brief Binds a descriptor set layout to this render stage.
+    /// @note This function will most likely be removed in the near future.
```

Comment:
```suggestion
    /// @deprecated This function will most likely be removed in the near future.
```

Reply:
The function isn't really deprecated yet. Deprecated would mean that the function has been superseded by another function and shouldn't be used anymore.

Reply:
A function must not be superseded in order to be deprecated. IMO a future removal is enough. If we're not really sure if we want to remove the function ('most likely') we can leave it as it is.

Reply:
In C++ you use the attribute deprecated to signal unstable or experimental function too. I think this is the right way to signal that.

Reply:
> A function must not be superseded in order to be deprecated. 

Eh, depends on the definition I guess :D

> In C++ you use the attribute deprecated to signal unstable or experimental function too

The deprecated attribute `Indicates that the name or entity declared with this attribute is deprecated, that is, the use is allowed, but discouraged for some reason`, but the use of this function isn't discouraged (it's the only way to use the API currently). If we mark the function as deprecated we will get spurious warnings from the compiler and clang-tidy.

On another note - I'm not sure why I changed it from "soon be removed" to "most likely be removed" since the function will definitely be removed at some point, I'm just not sure when exactly.

Reply:
Then let's change it to something like "This function will most likely be replaced by another function in the future". Then it is clear, that it is currently not discouraged but will be changed in the near future.

Reply:
Alright, I changed it to "will be removed" and added a bit more of a description as to why.